### ISSUE TITLE
Empty crock crashes client when interacting with clutter blocks

### ISSUE THREAD

**Issue by Agent00Ming** [anegostudios/VintageStory-Issues#3635]
### Game Version

1.19.4-rc1, 1.19.4-rc2, 1.19.4-rc3, 1.19.4

### Platform

Linux

### Modded

Vanilla

### SP/MP

Singleplayer

### Description

Interacting with any clutter blocks using an empty 'Fired Crock' causes the client to crash. It happens in both survival and creative. This would realistically happen when a player runs out of glue when repairing clutter blocks. (I missed this when testing #3588 on 1.19.4-rc1 and 1.19.4-rc2)

### How to reproduce

Locate any ruins with useless clutter blocks or grab some from the creative inventory and place them into the world. Equip an empty 'Fired Crock' and interact with the clutter block as if to repair it. Experience a crash.

### Logs
<details>
<summary>1.19.4-rc1</summary>

```
Game Version: v1.19.4-rc.1 (Unstable)
2/25/2024 4:05:23PM: Critical error occurred
Loaded Mods: game@1.19.4-rc.1, creative@1.19.4-rc.1, survival@1.19.4-rc.1
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.AssetLocation.ResolveToDomainAndPath(String domainAndPath, String& domain, String& path) in VintagestoryApi\Common\Assets\AssetLocation.cs:line 170
   at Vintagestory.GameContent.BlockBehaviorReparable.GetItemRepairAmount(IWorldAccessor world, ItemSlot slot) in VSSurvivalMod\BlockBehavior\BehaviorReparable.cs:line 211
   at Vintagestory.GameContent.BlockBehaviorReparable.OnBlockInteractStart(IWorldAccessor world, IPlayer byPlayer, BlockSelection blockSel, EnumHandling& handling) in VSSurvivalMod\BlockBehavior\BehaviorReparable.cs:line 121
   at Vintagestory.API.Common.Block.OnBlockInteractStart(IWorldAccessor world, IPlayer byPlayer, BlockSelection blockSel) in VintagestoryApi\Common\Collectible\Block\Block.cs:line 1346
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.TryBeginUseBlock(Block selectedBlock, BlockSelection blockSelection) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 864
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.HandleMouseInteractionsBlockSelected(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 532
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.UpdatePicking(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 220
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.OnFinalizeFrame(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 83
   at Vintagestory.API.Client.DummyRenderer.OnRenderFrame(Single deltaTime, EnumRenderStage stage) in VintagestoryApi\Client\API\IClientEventAPI.cs:line 83
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 200
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 804
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 1007
   at Vintagestory.Client.GuiScreenRunningGame.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 241
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 676
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 651
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 324
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
```
</details>
<details>
<summary>1.19.4-rc2</summary>

```
Running on 64 bit Linux (Arch Linux) [Kernel 6.7.6.1] with 31824 MB RAM
Game Version: v1.19.4-rc.2 (Unstable)
2/25/2024 4:06:07PM: Critical error occurred
Loaded Mods: game@1.19.4-rc.2, creative@1.19.4-rc.2, survival@1.19.4-rc.2
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.AssetLocation.ResolveToDomainAndPath(String domainAndPath, String& domain, String& path) in VintagestoryApi\Common\Assets\AssetLocation.cs:line 170
   at Vintagestory.GameContent.BlockBehaviorReparable.GetItemRepairAmount(IWorldAccessor world, ItemSlot slot) in VSSurvivalMod\BlockBehavior\BehaviorReparable.cs:line 211
   at Vintagestory.GameContent.BlockBehaviorReparable.OnBlockInteractStart(IWorldAccessor world, IPlayer byPlayer, BlockSelection blockSel, EnumHandling& handling) in VSSurvivalMod\BlockBehavior\BehaviorReparable.cs:line 121
   at Vintagestory.API.Common.Block.OnBlockInteractStart(IWorldAccessor world, IPlayer byPlayer, BlockSelection blockSel) in VintagestoryApi\Common\Collectible\Block\Block.cs:line 1346
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.TryBeginUseBlock(Block selectedBlock, BlockSelection blockSelection) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 864
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.HandleMouseInteractionsBlockSelected(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 532
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.UpdatePicking(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 220
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.OnFinalizeFrame(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 83
   at Vintagestory.API.Client.DummyRenderer.OnRenderFrame(Single deltaTime, EnumRenderStage stage) in VintagestoryApi\Client\API\IClientEventAPI.cs:line 83
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 200
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 806
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 1009
   at Vintagestory.Client.GuiScreenRunningGame.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 241
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 676
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 651
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 324
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
```
</details>
<details>
<summary>1.19.4-rc3</summary>

```
Running on 64 bit Linux (Arch Linux) [Kernel 6.7.6.1] with 31824 MB RAM
Game Version: v1.19.4-rc.3 (Unstable)
2/25/2024 4:06:55PM: Critical error occurred
Loaded Mods: game@1.19.4-rc.3, creative@1.19.4-rc.3, survival@1.19.4-rc.3
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.AssetLocation.ResolveToDomainAndPath(String domainAndPath, String& domain, String& path) in VintagestoryApi\Common\Assets\AssetLocation.cs:line 170
   at Vintagestory.GameContent.BlockBehaviorReparable.GetItemRepairAmount(IWorldAccessor world, ItemSlot slot) in VSSurvivalMod\BlockBehavior\BehaviorReparable.cs:line 217
   at Vintagestory.GameContent.BlockBehaviorReparable.OnBlockInteractStart(IWorldAccessor world, IPlayer byPlayer, BlockSelection blockSel, EnumHandling& handling) in VSSurvivalMod\BlockBehavior\BehaviorReparable.cs:line 121
   at Vintagestory.API.Common.Block.OnBlockInteractStart(IWorldAccessor world, IPlayer byPlayer, BlockSelection blockSel) in VintagestoryApi\Common\Collectible\Block\Block.cs:line 1357
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.TryBeginUseBlock(Block selectedBlock, BlockSelection blockSelection) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 864
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.HandleMouseInteractionsBlockSelected(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 532
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.UpdatePicking(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 220
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.OnFinalizeFrame(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 83
   at Vintagestory.API.Client.DummyRenderer.OnRenderFrame(Single deltaTime, EnumRenderStage stage) in VintagestoryApi\Client\API\IClientEventAPI.cs:line 83
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 200
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 806
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 1009
   at Vintagestory.Client.GuiScreenRunningGame.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 241
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 676
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 651
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 324
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
```
</details>

```Running on 64 bit Linux (Arch Linux) [Kernel 6.7.6.1] with 31824 MB RAM
Game Version: v1.19.4 (Stable)
2/28/2024 9:02:10AM: Critical error occurred
Loaded Mods: game@1.19.4, creative@1.19.4, survival@1.19.4
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.API.Common.AssetLocation.ResolveToDomainAndPath(String domainAndPath, String& domain, String& path) in VintagestoryApi\Common\Assets\AssetLocation.cs:line 170
   at Vintagestory.GameContent.BlockBehaviorReparable.GetItemRepairAmount(IWorldAccessor world, ItemSlot slot) in VSSurvivalMod\BlockBehavior\BehaviorReparable.cs:line 217
   at Vintagestory.GameContent.BlockBehaviorReparable.OnBlockInteractStart(IWorldAccessor world, IPlayer byPlayer, BlockSelection blockSel, EnumHandling& handling) in VSSurvivalMod\BlockBehavior\BehaviorReparable.cs:line 121
   at Vintagestory.API.Common.Block.OnBlockInteractStart(IWorldAccessor world, IPlayer byPlayer, BlockSelection blockSel) in VintagestoryApi\Common\Collectible\Block\Block.cs:line 1357
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.TryBeginUseBlock(Block selectedBlock, BlockSelection blockSelection) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 864
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.HandleMouseInteractionsBlockSelected(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 444
   at Vintagestory.Client.NoObf.SystemMouseInWorldInteractions.OnFinalizeFrame(Single dt) in VintagestoryLib\Client\Systems\Player\MouseInWorldInteractions.cs:line 89
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 186
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 809
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 1013
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 676
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 651
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 324
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
```

**Comment by radfast** [comment]
Thank you, much appreciated, we can fix this in time for `1.19.4-stable`

**Comment by Agent00Ming** [comment]
Can no longer replicate in 1.19.5-rc1
